tests/Queue/RedisQueueIntegrationTest.php

```diff
@@ -65,6 +65,49 @@ public function testExpiredJobsArePopped($driver)
         $this->assertEquals(3, $this->redis[$driver]->connection()->zcard('queues:default:reserved'));
     }
 
+    /**
+     * @dataProvider redisDriverProvider
+     *
+     * @param $driver
+     */
+    public function testMigrateMoreThan100Jobs($driver)
+    {
+        $this->setQueue($driver);
+
+        for ($i = -1; $i >= -201; $i--) {
+            $this->queue->later($i, new RedisQueueIntegrationTestJob($i));
+        }
+
+        for ($i = -201; $i <= -1; $i++) {
+            $this->assertEquals($i, unserialize(json_decode($this->queue->pop()->getRawBody())->data->command)->i);
+        }
+    }
+
+    /**
+     * @dataProvider redisDriverProvider
+     *
+     * @param $driver
+     *
+     * @throws \Exception
+     */
+    public function testBlockingPop($driver)
+    {
+        $this->tearDownRedis();
+        if ($pid = pcntl_fork() > 0) {
+            $this->setUpRedis();
+            $this->setQueue($driver, 10);
+            $this->assertEquals(12, unserialize(json_decode($this->queue->pop()->getRawBody())->data->command)->i);
+        } elseif ($pid == 0) {
+            $this->setUpRedis();
+            $this->setQueue('predis');
+            sleep(1);
+            $this->queue->push(new RedisQueueIntegrationTestJob(12));
+            die;
```

Comment:
?

Reply:
In order to test blocking pop, there should be a process (child) to push into queue while the worker process (parent) is waiting. After push the child process is done so it dies.

Reply:
If it doesn't matter, please use `exit("explaining why");`, think this is helpful for "the next developer". Alternatively, a comment in the source is welcome too. IMHO 😄